js/src/logger.ts

```diff
@@ -784,13 +890,16 @@ function updateSpanImpl({
       object_id: await parentObjectId.get(),
     }).objectIdFields();
 
+  const attachments = extractAttachments(updateEvent);
+
   const record = new LazyValue(async () => ({
     id,
     ...updateEvent,
     ...(await parentIds()),
     [IS_MERGE_FIELD]: true,
   }));
   state.bgLogger().log([record]);
+  state.bgLogger().logAttachments(attachments);
```

Comment:
Should we add the attachments parameter to `log` as a way to enforce that people don't forget about it? The logger is an internal-only API so it shouldn't be a problem to change the signature.

Reply:
Done